### ISSUE TITLE
[Docs] Exact GP with Pyro

### ISSUE THREAD

**Issue by tawe141** [cornellius-gp/gpytorch#1628]
Something strikes me as a bit funny in the [docs about fully Bayesian GPs](https://docs.gpytorch.ai/en/latest/examples/01_Exact_GPs/GP_Regression_Fully_Bayesian.html). The `pyro_model` is defined as such: 

```
def pyro_model(x, y):
    model.pyro_sample_from_prior()
    output = model(x)
    loss = mll.pyro_factor(output, y)
    return y
```
The model returns the exact y even though the function computes the output and the loss, neither of which are returned but are calculated anyway. Is this intentional, or is something wrong here?

**Comment by jacobrgardner** [comment]
Returning y is not necessary, but computing the output and the loss is. The key is the pyro_factor call, which let's pyro's internals know we want to use the mll (plus log priors) as a potential energy function for sampling.

Note though that pyro sampling is a bit bugged until #1594 goes in. That PR is ready though, so should go in soon now that we're done with the NeurIPS deadline.
